satpy/tests/test_demo.py

```diff
@@ -192,3 +192,72 @@ def test_ahi_partial_download(self):
         from tempfile import gettempdir
         files = download_typhoon_surigae_ahi(base_dir=gettempdir(), segments=[4, 9], channels=[1, 2, 3])
         assert len(files) == 6
+
+
+class _FakeRequest:
+    """Fake object to act like a requests return value when downloading a zip file."""
+
+    def __init__(self, url, stream=None):
+        self._filename = os.path.basename(url)
+        self.headers = {}
+        del stream  # just mimicking requests 'get'
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        return
+
+    def raise_for_status(self):
+        return
+
+    def _get_fake_viirs_sdr_bytesio(self):
+        filelike_obj = io.BytesIO()
+        filelike_obj.write(b"1234")
+        filelike_obj.seek(0)
+        return filelike_obj
+
+    def iter_content(self, chunk_size):
+        """Return generator of 'chunk_size' at a time."""
+        bytes_io = self._get_fake_viirs_sdr_bytesio()
+        x = bytes_io.read(chunk_size)
+        while x:
+            yield x
+            x = bytes_io.read(chunk_size)
+
+
+class TestVIIRSSDRDemoDownload:
+    """Test VIIRS SDR downloading."""
+
+    @mock.patch('satpy.demo.viirs_sdr.requests')
+    def test_download(self, _requests, tmpdir):
+        """Test downloading and re-downloading VIIRS SDR data."""
+        from satpy.demo import get_viirs_sdr_20170128_1229
+        _requests.get.side_effect = _FakeRequest
+        files = get_viirs_sdr_20170128_1229(base_dir=str(tmpdir))
+        assert len(files) == 10 * (16 + 5 + 1 + 3)  # 10 granules * (5 I bands + 16 M bands + 1 DNB + 3 geolocation)
```

Comment:
You could also check the contents of the files...

Reply:
Filenames or contents of the files?

Reply:
I really meant contents, but filenames would be good too.

Reply:
In this case though the contents are going to all be the same for my tests. Ugh fine I'll do it the better way.